test/e2e/e2e_test.go

```diff
@@ -115,6 +115,7 @@ func TestMain(m *testing.M) {
 	flag.StringVar(&globalOpts.ManagementClusterNamespace, "e2e.management-cluster-namespace", "", "Namespace of the management cluster's HostedCluster (required to test request serving isolation)")
 	flag.StringVar(&globalOpts.ManagementClusterName, "e2e.management-cluster-name", "", "Name of the management cluster's HostedCluster (required to test request serving isolation)")
 	flag.BoolVar(&globalOpts.DisablePKIReconciliation, "e2e.disable-pki-reconciliation", false, "If set, TestUpgradeControlPlane will upgrade the control plane without reconciling the pki components")
+	flag.StringVar(&globalOpts.configurableClusterOptions.Annotation, "e2e.annotation", "", "An annotation to apply to the HostedCluster (key=value)")
```

Comment:
I'd suggest making the annotations repeatable.  That way we can add more than one annotation at a time. That will probably help us out in the future.

you can achieve this using a custom var with the flags package. i tested this out. here's an example.

```
package main

import (
        "flag" 
        "fmt"
        "strings"
)

type stringMapVar map[string]string

func (s *stringMapVar) String() string {
        if *s == nil {
                return ""
        }
        return fmt.Sprintf("%v", *s)
}

func (s *stringMapVar) Set(value string) error {
        split := strings.Split(value, "=")
        if len(split) != 2 {
                return fmt.Errorf("invalid argument: %s", value)
        }
        if *s == nil {
                *s = stringMapVar(map[string]string{})
        }
        map[string]string(*s)[split[0]] = split[1]
        return nil
}

var annotations stringMapVar

func main() {

        flag.Var(&annotations, "annotations", "")
        flag.Parse()

        if annotations != nil {
                fmt.Printf("%s\n", annotations.String())
        }
}

```

Reply:
done, thanks.
`e2e.annotations` can be specified multiple times and the annotations are aggregated.